xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -164,19 +196,36 @@ public WeightedChildLbState(Object key, LoadBalancerProvider policyProvider, Obj
     }
 
     private double getWeight() {
-      if (config == null) {
-        return 0;
-      }
-      long now = ticker.nanoTime();
-      if (now - lastUpdated >= config.weightExpirationPeriodNanos) {
-        nonEmptySince = infTime;
-        return 0;
-      } else if (now - nonEmptySince < config.blackoutPeriodNanos
-          && config.blackoutPeriodNanos > 0) {
-        return 0;
-      } else {
-        return weight;
+      double weightToReturn = 0;
+      if (config != null) {
+        long now = ticker.nanoTime();
+        if (weightIsStale(now)) {
+          nonEmptySince = infTime;
+          // TODO: add target and locality labels once available
+          endpointWeightStaleCounter.add(1, ImmutableList.of(), ImmutableList.of());
```

Comment:
I'm not at all wild about recording metrics here; this is very deep and on a getter. It also records individually for each endpoint.

I'm not in love with it, but better would be to update the metrics in `updateWeight()`. It already has access to the weights, and passing in two (Non-)AtomicInteger counters to this getWeight() would allow accumulating to update them once after looping through all endpoints. Looks like that is [what was done in C](https://github.com/grpc/grpc/commit/40577dd585274aa9d0f78682626d55e402db9003#diff-12d07e1fc327615a2e72ef697950c64bc8fc68acf306f34f9086c3eb3a990914R1462).